### DISCUSSION TITLE
Welcome to View-Assist Discussions!

### DISCUSSION THREAD

**Discussion by dinki** [dinki/View-Assist#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by pgale** [comment]
Thanks for opening up discussions - I was aware the issues area wasn't the best for my previous discussion!

**Reply by dinki** [reply]
No worries.  I hadn't thought about it until I saw your post in issues and realized that people may want to talk about things.

**Comment by pgale** [comment]
One question - has your work been discussed in a specific thread on other forums? I did have a search last week but couldn't see anything obvious. Thanks.

**Reply by dinki** [reply]
The only thread available now is the contest entry.  I am hoping to get things tidied up and up here on GH and then I'll start a thread on the HA forums in the 'Share your project' section.  I know those threads can get long over time and make it difficult to find things.  Maybe this discussion area will allow for some focused conversation threads rather than one big one.

**Comment by DJS91** [comment]
Hey dinki, I saw your demo has an auditory acknowledgement after wake word detection in your award winning assistant! 
Where can I view the code for this function specifically? This is exactly what I've been looking for.

**Reply by dinki** [reply]
This is part of the Stream Assist configuration.  Take a look at the video on this page:
https://github.com/dinki/View-Assist/wiki/StreamAssist-installation-and-configuration

**Reply by OzoneActionDay** [reply]
@DanStennett, I had that same question when I was setting up my environment.   I was able to play the chime on my tablet but I couldn't figure out how to trigger it when the wake word was heard.   I'm not sure if this is the right way but this is how I set things up.

On line 4 of dinki's example yaml for the clock card he references a binary sensor named "input_boolean.lr_tablet_mic".    I figured that must be how the card knows that the wake word had been heard and that the chime needed to be called.    I couldn't find that binary sensor anywhere in my environment so I created one by hand which keys off of the "WAKE" entity that's in my Stream Assist integration.  That entity's state is normally "start" but when it hears the wake word it changes to "end" so I used that change as a trigger to change the state of my binary sensor.

In my case the name of my tablet in Stream Assist is "Galaxy Tab" and I've renamed that binary sensor from "lr_tablet_mic" to "galaxy_tab_wake_word_triggered" .  Here's how I've defined that entity in my yaml:

```
- platform: template
  sensors:
    galaxy_tab_wake_word_triggered:
     unique_id: galaxy_tab_wake_word_triggered
     friendly_name: Galaxy Tab Wake Word Triggered
     value_template: >-
       {% if (states('sensor.galaxy_tab_wake') == "end") %}
         1
       {% else %}
         0
       {% endif %}

```

I don't know if I missed a step where this would've been done automagically so I'm interested to hear how other people have set this up.

Also, hats off to @dinki for coming up with this solution.  It takes a special talent to be able to look at the tools that are out there and to figure out how to tie them together to make something awesome.

**Reply by dinki** [reply]
Thanks for the kind words @ekashuba .  I was able to leverage a bunch of help from the good folks on Discord to be able to piece a lot of this together.    Thanks also for everyone's patience.  I'm in the process of a major revision that HOPEFULLY will make things easier to deploy and not hard coded like the pieces you are referencing.  Until then, know that what I have posted is functional but will be superseded by something with a lot more features.

I had not realized that I did not provide the automation that does the detect until you posted your message.  That's a big oversight.  To be honest, it's been about a month since I've rewritten things so it's hard to remember how the original proof of concept was written but you are on the right path with your solution or at least on the same path I use.  

Here's an example of what that looks like now:

template.yaml
```
  - name: Assist_Last_Called
    state: >-
      {%- set sensors = expand(['group.assist_mics'])|selectattr('state', 'eq', 'start')|rejectattr('last_changed', 'eq', None)|list -%}
      {{ sensors|max(attribute="last_changed")|attr('entity_id') if sensors|length >= 1 else 'None'}}
```

This template sensor looks through a group of View Assist mic devices, checks for the 'start' state of the mic and then finds the one that was last changed.   So what this accomplishes is returning what microphone was used.  From there, you can then figure out the custom device's media player and browser display.  Those are used in the automations and custom sentences.   So far this works really well.

I'm also working on incorporating voice only satellites using on device wake word.  This will allow for all Assist devices to be 'managed' by View Assist and allow for some features like setting alarms/reminders, playing music, communication between two or more devices (planned), do not disturb mode, and a few other ideas.   I am working through getting those in currently and then I think the core of what I want to accomplish for my set up will be done.
